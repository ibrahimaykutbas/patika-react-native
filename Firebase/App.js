import React from 'react';
import {View, Text, TouchableOpacity, ScrollView} from 'react-native';

import auth from '@react-native-firebase/auth';

import database from '@react-native-firebase/database';

const App = () => {
  const button = (title, process) => {
    return (
      <TouchableOpacity
        style={{
          backgroundColor: '#e3f2fd',
          width: 200,
          height: 50,
          justifyContent: 'center',
          alignItems: 'center',
          alignSelf: 'center',
          borderRadius: 5,
          marginBottom: 10,
          color: 'black',
        }}
        onPress={process}>
        <Text style={{color: 'black'}}>{title}</Text>
      </TouchableOpacity>
    );
  };

  const signUp = () => {
    auth()
      .createUserWithEmailAndPassword(
        'jane.doe@example.com',
        'SuperSecretPassword!',
      )
      .then(() => console.log(res))
      .catch(err => console.log(err));
  };

  const signIn = () => {
    auth()
      .signInWithEmailAndPassword(
        'jane.doe@example.com',
        'SuperSecretPassword!',
      )
      .then(() => console.log('Signed in..'))
      .catch(err => console.log(err));
  };

  const signOut = () => {
    auth()
      .signOut()
      .then(() => console.log('Signed out..'))
      .catch(err => console.log(err));
  };

  const checkOut = () => {
    const user = auth().currentUser;
    console.log(user);
  };

  const checkDB = () => {
    const reference = database().ref();
    reference.once('value').then(snapshot => {
      const response = snapshot.val();
      console.log(response);
    });
  };

  const listenDB = () => {
    const reference = database().ref('');
    reference.on('value', snapshot => {
      console.log(snapshot.val());
    });
  };

  const setDB = () => {
    const reference = database().ref('car/rentable');
    reference.set({
      brand: 'Audi',
      model: 'A8',
      price: 128,
    });
  };

  const updateDB = () => {
    const reference = database().ref('car/rentable');
    reference.update({
      model: 'A3',
    });
  };

  const pushDB = () => {
    const reference = database().ref('car');
    reference.push({
      brand: 'Passat',
      model: '81',
      price: 650,
    });
  };

  const removeDB = () => {
    const reference = database().ref('car');
    reference.remove();
  };

  return (
    <View style={{backgroundColor: 'white', flex: 1}}>
      <Text
        style={{
          color: 'black',
          fontWeight: 'bold',
          textAlign: 'center',
          marginTop: 100,
          marginBottom: 10,
        }}>
        Transactions
      </Text>
      <ScrollView showsHorizontalScrollIndicator={false}>
        {button('Sign Up', signUp)}
        {button('Sign In', signIn)}
        {button('Sign Out', signOut)}
        {button('Check User', checkOut)}
        {button('Check DB', checkDB)}
        {button('Listen DB', listenDB)}
        {button('Set DB', setDB)}
        {button('Update DB', updateDB)}
        {button('Push DB', pushDB)}
        {button('Remove DB', removeDB)}
      </ScrollView>
    </View>
  );
};

export default App;
