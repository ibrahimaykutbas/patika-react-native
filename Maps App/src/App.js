import React, {useRef, useState} from 'react';
import {View, Text} from 'react-native';

import MapView from 'react-native-maps';

import UserMarker from './components/marker/UserMarker/UserMarker';
import Loading from './components/Loading/Loading';
import InfoCard from './components/InfoCard/InfoCard';

import useFetch from './hooks/useFetch';

const App = () => {
  const mapRef = useRef();

  const [user, setUser] = useState();
  const [modalVisibility, setModalVisibility] = useState(false);

  const {data, loading, error} = useFetch(
    'https://random-data-api.com/api/users/random_user?size=50',
  );

  const renderUserMarker = () => {
    return data.map(
      ({
        id,
        first_name,
        last_name,
        username,
        avatar,
        address: {coordinates},
      }) => {
        return (
          <UserMarker
            key={id}
            coordinates={{
              latitude: coordinates.lat,
              longitude: coordinates.lng,
            }}
            userImageURL={avatar}
            onPress={() =>
              handleMarkerSelect(coordinates, {
                first_name,
                last_name,
                username,
              })
            }
          />
        );
      },
    );
  };

  function handleMarkerSelect(coor, selectedUser) {
    setUser(selectedUser);
    handleModalVisibility();
    mapRef.current.animateToRegion({
      latitude: coor.lat,
      longitude: coor.lng,
      latitudeDelta: 10,
      longitudeDelta: 10,
    });
  }

  function handleModalVisibility() {
    setModalVisibility(!modalVisibility);
  }

  return (
    <View style={{flex: 1}}>
      <MapView ref={mapRef} style={{flex: 1}}>
        {data && renderUserMarker()}
      </MapView>
      {loading && <Loading />}
      {user && (
        <InfoCard
          user={user}
          visible={modalVisibility}
          close={handleModalVisibility}
        />
      )}
    </View>
  );
};

export default App;
