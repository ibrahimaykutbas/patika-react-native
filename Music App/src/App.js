import React from 'react';
import {StyleSheet, View, FlatList} from 'react-native';

import musicList from './db.json';

import SongCard from './components/SongCard/SongCard';
import Header from './components/Header';

const App = () => {
  const renderSong = ({item}) => <SongCard song={item} />;
  const renderSeperator = () => <View style={styles.seperator} />;
  return (
    <View style={styles.container}>
      <FlatList
        ListHeaderComponent={<Header />}
        keyExtractor={item => item.id}
        data={musicList}
        renderItem={renderSong}
        ItemSeparatorComponent={renderSeperator}
      />
    </View>
  );
};

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  seperator: {
    borderWidth: 1,
    borderColor: '#e0e0e0',
  },
});
