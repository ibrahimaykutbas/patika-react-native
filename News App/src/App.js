import React from 'react';
import {SafeAreaView, View, FlatList, StyleSheet, Text} from 'react-native';

import news_data from './news_data.json';

import NewsCard from './components/NewsCard';
import NewsBanner from './components/NewsBanner';

function App() {
  const renderNews = ({item}) => <NewsCard news={item} />;
  const keyNew = item => item.u_id.toString();

  return (
    <SafeAreaView style={styles.container}>
      <View>
        <Text style={styles.headerText}>News App </Text>
        <FlatList
          ListHeaderComponent={() => <NewsBanner />}
          keyExtractor={keyNew}
          data={news_data}
          renderItem={renderNews}
        />
      </View>
    </SafeAreaView>
  );
}
const styles = StyleSheet.create({
  container: {flex: 1, backgroundColor: '#ECEFF1'},
  headerText: {
    fontWeight: 'bold',
    fontSize: 40,
  },
});

export default App;
