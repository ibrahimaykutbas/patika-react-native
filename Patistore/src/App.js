import React from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';

import Header from './components/Header';
import Content from './components/Content';

function App() {
  return (
    <SafeAreaView style={styles.container}>
      <Header />
      <Content />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFAFF',
  },
});

export default App;
