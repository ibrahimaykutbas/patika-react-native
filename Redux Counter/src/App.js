import React from 'react';
import {View, Text, Button} from 'react-native';

import {createStore} from 'redux';
import {Provider, useSelector, useDispatch} from 'react-redux';

const initialState = {
  counter: 0,
};

const reducers = (state, action) => {
  switch (action.type) {
    case 'INCREASE_COUNTER':
      return {...state, counter: state.counter + 1};

    case 'DECREASE_COUNTER':
      return {...state, counter: state.counter - 1};

    default:
      return state;
  }
};

export default () => {
  const store = createStore(reducers, initialState);
  return (
    <Provider store={store}>
      <View style={{flex: 1}}>
        <First />
        <Second />
      </View>
    </Provider>
  );
};

const First = () => {
  const counter = useSelector(selector => selector.counter);
  const dispatch = useDispatch();

  const handleIncrease = () => {
    dispatch({type: 'INCREASE_COUNTER'});
  };
  return (
    <View style={{flex: 1, backgroundColor: '#eceff1'}}>
      <Text style={{fontSize: 30, color: 'black'}}>First: {counter}</Text>
      <Button title="Increase" onPress={handleIncrease} />
    </View>
  );
};

const Second = () => {
  const counter = useSelector(selector => selector.counter);
  const dispatch = useDispatch();

  const handleDecrease = () => {
    counter > 0 && dispatch({type: 'DECREASE_COUNTER'});
  };
  return (
    <View style={{flex: 1}}>
      <Text style={{fontSize: 30, color: 'black'}}>Second: {counter}</Text>
      <Button title="Decrease" onPress={handleDecrease} />
    </View>
  );
};
