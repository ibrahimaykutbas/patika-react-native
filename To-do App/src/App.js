import React, {useState} from 'react';
import {StyleSheet, Text, View} from 'react-native';

import Header from './components/Header/Header';
import Todos from './components/Todos/Todos';

const App = () => {
  const [count, setCount] = useState(0);

  const getTodos = todos => {
    const filtered = todos.filter(todo => todo.status === false);
    setCount(filtered.length);
  };

  return (
    <View style={styles.container}>
      <Header count={count} />
      <Todos getTodos={getTodos} />
    </View>
  );
};

export default App;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirector: 'row',
    backgroundColor: '#334756',
  },
});
