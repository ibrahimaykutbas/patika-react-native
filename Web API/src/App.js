import React, {useState, useEffect} from 'react';
import {View, ActivityIndicator, FlatList} from 'react-native';

import axios from 'axios';

import UserCard from './components/UserCard/UserCard';

const App = () => {
  const [loading, setLoading] = useState(true);
  const [userList, setUserList] = useState([]);

  async function getData() {
    const url = 'https://jsonplaceholder.typicode.com/users';
    const response = await axios.get(url);
    setLoading(false);
    setUserList(response.data);

    /* axios.get(url).then(response => {
      setLoading(false);
      setUserList(response.data);
    }); */
  }

  useEffect(() => {
    getData();
  }, []);

  const renderUser = ({item}) => <UserCard user={item} />;

  return (
    <View>
      {loading ? (
        <ActivityIndicator size="large" />
      ) : (
        <FlatList data={userList} renderItem={renderUser} />
      )}
    </View>
  );
};

export default App;
