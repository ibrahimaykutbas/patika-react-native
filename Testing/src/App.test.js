import React from 'react';

import {render, fireEvent} from '@testing-library/react-native';

import App from './App';

test('should match with snapshot', () => {
  const comp = render(<App />);
  expect(comp).toMatchSnapshot();
});

test('should add item to list', () => {
  const comp = render(<App />);

  const input = comp.getByTestId('input-area');
  fireEvent.changeText(input, 'test element');

  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(1);
});

test('should add item for multiple times', () => {
  const comp = render(<App />);

  const input = comp.getByTestId('input-area');
  fireEvent.changeText(input, 'test element');

  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);
  fireEvent.press(submitButton);
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(3);
});

test('should not add item if input area is empty', () => {
  const comp = render(<App />);

  const submitButton = comp.getByTestId('button-touchable');
  fireEvent.press(submitButton);

  const list = comp.getByTestId('list').props;
  expect(list.data.length).toBe(0);
});
