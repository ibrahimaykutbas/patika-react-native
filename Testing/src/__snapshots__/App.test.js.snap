// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <Text>
    Testing
  </Text>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="list"
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <TextInput
    onChangeText={[Function]}
    placeholder="Add..."
    style={
      {
        "backgroundColor": "#e0e0e0",
        "borderRadius": 5,
        "margin": 10,
        "padding": 10,
      }
    }
    testID="input-area"
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 10,
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="button-touchable"
  >
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
        }
      }
      testID="button-title"
    >
      Add
    </Text>
  </View>
</RCTSafeAreaView>
`;
